Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs

```diff
@@ -598,5 +574,39 @@ private bool IsBestSubmission(int problemId, int participantId, int submissionId
 
             return bestScore?.SubmissionId == submissionId;
         }
+
+        private void RetestSubmission(DatabaseModelType submission)
+        {
+            if (!submission.ProblemId.HasValue || !submission.ParticipantId.HasValue)
+            {
+                return;
```

Comment:
This is a validity check, we cannot simply `return` if it fails. An appropriate error message should be returned and displayed to the user instead.